dotnet/samples/Demos/OnnxSimpleRAG/README.md

```diff
@@ -0,0 +1,86 @@
+- [Text Embeddings Generation Service]() - Using the Text Embeddings Generation Service from [Onnx Connector](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.Onnx/BertOnnxTextEmbeddingGenerationService.cs) to generate
+- [Memory Store](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/Memory/IMemoryStore.cs) Using Memory Store Service with [VolatileMemoryStore](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Plugins/Plugins.Memory/VolatileMemoryStore.cs) to store and retrieve embeddings in memory for RAG.
+- [Semantic Text Memory](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Core/Memory/SemanticTextMemory.cs) to manage the embeddings in memory for RAG.
+- [Text Memory Plugin](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Plugins/Plugins.Memory/TextMemoryPlugin.cs) to enable memory retrieval functions (Recall) to be used with Prompts for RAG.
+
+## Prerequisites
+
+- [.NET 8](https://dotnet.microsoft.com/download/dotnet/8.0).
+
+## 1. Configuring the sample
+
+### Downloading the Models
+
+For this example we chose Hugging Face as our repository for download of the local models, go to a directory of your choice where the models should be downloaded and run the following commands:
+
+```powershell
+git clone https://huggingface.co/TaylorAI/bge-micro-v2
+git clone https://huggingface.co/microsoft/Phi-3-mini-4k-instruct-onnx
+```
+
+> [!IMPORTANT]
+> Both `BG2` and `Phi-3` models are too large to be downloaded by the `git clone` command alone if you don't have [git-lfs extension](https://git-lfs.com/) installed, for this you may need to download the models manually and overwrite the files in the cloned directories.
+
+- Manual download [BGE-V2](https://huggingface.co/TaylorAI/bge-micro-v2/resolve/main/onnx/model.onnx?download=true) (69 MB)
+- Manual download [Phi-3 CPU](https://huggingface.co/microsoft/Phi-3-mini-4k-instruct-onnx/resolve/main/cpu_and_mobile/cpu-int4-rtn-block-32/phi3-mini-4k-instruct-cpu-int4-rtn-block-32.onnx.data?download=true) (≈2.7 GB)
+
+## 2. Configuring the sample
+
+Update the `Program.cs` file lines below with the paths to the models you downloaded in the previous step.
+
+```csharp
+// Path to the folder of your downloaded ONNX PHI-3 model
+var chatModelPath = @"C:\path\to\huggingface\Phi-3-mini-4k-instruct-onnx\cpu_and_mobile\cpu-int4-rtn-block-32";
+
+// Path to the file of your downloaded ONNX BGE-MICRO-V2 model
+var embeddingModelPath = @"C:\path\to\huggingface\bge-micro-v2\onnx\model.onnx";
+
+// Path to the vocab file your ONNX BGE-MICRO-V2 model
+var embeddingVocabPath = @"C:\path\to\huggingface\bge-micro-v2\vocab.txt";
+```
+
+**Optional**: Change or add any fact text you want your AI to know about to the `facts` list in the `Program.cs` file.
+
+```csharp
+foreach (var fact in new[] {
+    "My fact 1.",
+    "My fact 2." })
+{
```

Comment:
Would be nice to include this as an external .txt file

Reply:
I can do something that you provide a folder and traverse the `.txt` files as facts, good suggestion.